### PR TITLE
Adding custom ME management port for scp transfer

### PR DESCRIPTION
**Pull request by vmonnier**

### CODE REVIEW

File: adapters/catalyst_ios/common.php [hunk 1]
```diff
@@ -176,8 +176,9 @@ function scp_to_router($src, $dst)
 	$net_profile = get_network_profile();
 	$sd = &$net_profile->SD;
 	$sd_ip_addr = $sd->SD_IP_CONFIG;
+        $sd_mgt_port = $sd->SD_MANAGEMENT_PORT;
 
-	$ret_scp = exec_local(__FILE__.':'.__LINE__, "/opt/sms/bin/sms_scp_transfer -s $src -d /$dst -l $login -a $sd_ip_addr -p $passwd", $output);
+	$ret_scp = exec_local(__FILE__.':'.__LINE__, "/opt/sms/bin/sms_scp_transfer -s $src -d /$dst -l $login -a $sd_ip_addr -p $passwdi -P $sd_mgt_port", $output);
```
Comment (UBiqube-ydu) [inline adapters/catalyst_ios/common.php:181]: $passwdi ==> $passwd

File: adapters/cisco_asa_rest/common.php [hunk 2]
```diff
@@ -332,8 +333,9 @@ function scp_to_router($src, $dst)
   $net_profile = get_network_profile();
   $sd = &$net_profile->SD;
   $sd_ip_addr = $sd->SD_IP_CONFIG;
+  $sd_mgt_port = $sd->SD_MANAGEMENT_PORT;
 
-  $ret_scp = exec_local(__FILE__ . ':' . __LINE__, "/opt/sms/bin/sms_scp_transfer -s $src -d $dst_disk:/$dst -l $login -a $sd_ip_addr -p $passwd", $output);
+  $ret_scp = exec_local(__FILE__ . ':' . __LINE__, "/opt/sms/bin/sms_scp_transfer -s $src -d $dst_disk:/$dst -l $login -a $sd_ip_addr -p $passwdi -P $sd_mgt_port", $output);
```
Comment (UBiqube-ydu) [inline adapters/cisco_asa_rest/common.php:338]: $passwdi ==> $passwd

**Review by Vignaudo — APPROVED** [review]

File: adapters/docker_generic/common.php [hunk 3]
```diff
@@ -146,8 +147,9 @@ function scp_to_router($src, $dst)
   $sd_ip_addr = $sd->SD_IP_CONFIG;
   $login = $sd->SD_LOGIN_ENTRY;
   $passwd = $sd->SD_PASSWD_ENTRY;
+  $sd_mgt_port = $sd->SD_MANAGEMENT_PORT;
 
-  $ret_scp = exec_local(__FILE__ . ':' . __LINE__, "/opt/sms/bin/sms_scp_transfer -s $src -d $dst_disk/$dst -l $login -a $sd_ip_addr -p '$passwd' ", $output);
+  $ret_scp = exec_local(__FILE__ . ':' . __LINE__, "/opt/sms/bin/sms_scp_transfer -s $src -d $dst_disk/$dst -l $login -a $sd_ip_addr -p '$passwd -P $sd_mgt_port' ", $output);
```
Comment (UBiqube-ydu) [inline adapters/docker_generic/common.php:152]: `-p '$passwd -P $sd_mgt_port'` ==> `-p '$passwd' -P $sd_mgt_port`
Be carefull to the simple quote around passwd!

File: adapters/juniper_srx/juniper_srx_configuration.php [hunk 4]
```diff
@@ -408,10 +409,11 @@ function update_license()
     $ipaddr = $sms_sd_ctx->getIpAddress();
     $login = $sms_sd_ctx->getLogin();
     $passwd = $sms_sd_ctx->getPassword();
+    $sd_mgt_port = $sd->SD_MANAGEMENT_PORT;
```
Comment (UBiqube-ydu) [inline adapters/juniper_srx/juniper_srx_configuration.php:412]: `$sd` is not initialized

File: adapters/veex_rtu/common.php [hunk 5]
```diff
@@ -282,8 +282,9 @@ function scp_from_router($src, $dst)
   $net_profile = get_network_profile();
   $sd = &$net_profile->SD;
   $sd_ip_addr = $sd->SD_IP_CONFIG;
+  $sd_mgt_port = $sd->SD_MANAGEMENT_PORT;
 
-  $ret_scp = exec_local(__FILE__ . ':' . __LINE__, "/opt/sms/bin/sms_scp_transfer -r -s $src -d $dst -l $login -a $sd_ip_addr -p $passwd", $output);
+  $ret_scp = exec_local(__FILE__ . ':' . __LINE__, "/opt/sms/bin/sms_scp_transfer -r -s $src -d $dst -l $login -a $sd_ip_addr -p $passwd -P $sd_mgt_por", $output);
```
Comment (UBiqube-ydu) [inline adapters/veex_rtu/common.php:287]: `$sd_mgt_por` ==>  `$sd_mgt_port`

File: adapters/wsa/wsa_configuration.php [hunk 6]
```diff
@@ -162,9 +162,12 @@ function get_running_conf()
     $login = $sms_sd_ctx->getLogin();
     $passwd = $sms_sd_ctx->getPassword();
 
+    $SD = &$network->SD;
```
Comment (UBiqube-ydu) [inline adapters/wsa/wsa_configuration.php:165]: `$network` is not initialized.

File: adapters/catalyst_ios/common.php (hunk 1 above)
Comment (vmonnier) [reply adapters/catalyst_ios/common.php]: fix by this commit https://github.com/openmsa/Adapters/pull/295/commits/38a1123dd0717365f493b27ae08b7561d111d3c8

File: adapters/cisco_asa_rest/common.php (hunk 2 above)
Comment (vmonnier) [reply adapters/cisco_asa_rest/common.php]: fix by this commit https://github.com/openmsa/Adapters/pull/295/commits/38a1123dd0717365f493b27ae08b7561d111d3c8

File: adapters/docker_generic/common.php (hunk 3 above)
Comment (vmonnier) [reply adapters/docker_generic/common.php]: fix by this commit https://github.com/openmsa/Adapters/pull/295/commits/38a1123dd0717365f493b27ae08b7561d111d3c8

File: adapters/juniper_srx/juniper_srx_configuration.php (hunk 4 above)
Comment (vmonnier) [reply adapters/juniper_srx/juniper_srx_configuration.php:416]: fixed by this commit https://github.com/openmsa/Adapters/pull/295/commits/d5ee499d2ebe0b444954df206cd7458cbd28e3e1

File: adapters/veex_rtu/common.php (hunk 5 above)
Comment (vmonnier) [reply adapters/veex_rtu/common.php]: fix by this commit https://github.com/openmsa/Adapters/commit/38a1123dd0717365f493b27ae08b7561d111d3c8

File: adapters/wsa/wsa_configuration.php (hunk 6 above)
Comment (vmonnier) [reply adapters/wsa/wsa_configuration.php:166]: fix by this commit https://github.com/openmsa/Adapters/commit/38a1123dd0717365f493b27ae08b7561d111d3c8

File: adapters/juniper_srx/juniper_srx_configuration.php [hunk 7]
```diff
@@ -149,10 +149,11 @@ function update_firmware($param = '')
 
     $ipaddr = $sms_sd_ctx->getIpAddress();
     $login = $sms_sd_ctx->getLogin();
+    $sd_mgt_port = $sd->SD_MANAGEMENT_PORT;
```
Comment (UBiqube-ydu) [inline adapters/juniper_srx/juniper_srx_configuration.php:152]: $sd --> $SD

File: adapters/juniper_srx/juniper_srx_configuration.php (hunk 4 above)
Comment (UBiqube-ydu) [reply adapters/juniper_srx/juniper_srx_configuration.php:416]: line 416 should not be there

Comment (vmonnier) [reply adapters/juniper_srx/juniper_srx_configuration.php]: fixed by commit  [4c21b11](https://github.com/openmsa/Adapters/pull/295/commits/4c21b11278e410b25f66da4800d8b072047a4202)

**Pull request merged**